src/Gui/View3DInventor.cpp

```diff
@@ -513,13 +513,19 @@ void View3DInventor::printPreview()
 {
     QPrinter printer(QPrinter::ScreenResolution);
     printer.setFullPage(true);
-    printer.setPageSize(QPageSize(QPageSize::A4));
-    printer.setPageOrientation(QPageLayout::Landscape);
+    hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/View");
+    QPageSize::PageSizeId initialDefaultPageSize = QLocale::system().country() == QLocale::UnitedStates ? QPageSize::Letter : QPageSize::A4;

```

Comment:
You could consider using `QPrinter().pageSize()` instead of the condition here: it relies on the paper size set by the system's default printer, and would handle cases such as Canada, for example, where I think you find both A4 and Letter paper sizes with some frequency.

Reply:
Okay.  Yes, that would be better to get the defaults from the printer.  I didn't have a printer setup in my virtual machine, so I added one and did it this way.  Your suggestion didn't work, so I used this method with QPrinterInfo class instead.